packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte

```diff
@@ -151,6 +150,15 @@
 		text = "";
 		error = "";
 		try {
+			const opts = {
+				dont_load_model: isOnLoadCall,
+				includeCredentials,
+				signal: abort?.signal,
+				use_cache: useCache,
+				wait_for_model: withModelLoading,
+			} satisfies Options;
+
+			tgiSupportedModels = await getTgiSupportedModels(apiUrl);
```

Comment:
why removing this from `onMount`? I would assume the list never change.

Reply:
The main reason is that there is a race condition between `getTgiSupportedModels` and `applyWidgetExamples` on load.

The latter is called `onMount` by `WidgetExample`:
https://github.com/huggingface/huggingface.js/blob/d9c933a4611ddd55722a4eb39cbd95eb66ec205f/packages/widgets/src/lib/components/InferenceWidget/shared/WidgetExamples/WidgetExamples.svelte#L94-L117

Which would result in `ConversationalWidget.getOutput` being called before `getTgiSupportedModels` has completed. As a consequence, the example showed on load is not using the TGI streaming API although it should.


The change introduced in this PR makes sure `getTgiSupportedModels` is complete before proceeding with the inference call.
Note that calling `getTgiSupportedModels` several times in a row does not trigger more than one network call to fetch the info from the inference API.

------

>  I would assume the list never change.
This is mostly true during the inference widget's lifespan, but not absolutely true. The list represents models that are currently deployed with TGI in the inference API and can change over time.

Reply:
> The change introduced in this PR makes sure getTgiSupportedModels is complete before proceeding with the inference call. 

Thanks for the explanation!

> Note that calling getTgiSupportedModels several times in a row does not trigger more than one network call to fetch the info from the inference API.

Oh great! Then it's all good :+1: 

> This is mostly true during the inference widget's lifespan, but not absolutely true. 

Yes indeed, I was talking about within the inference widget lifespan as well :+1: